nomad/structs/structs_test.go

```diff
@@ -6024,3 +5917,30 @@ func TestTaskGroup_validateScriptChecksInGroupServices(t *testing.T) {
 		require.Nil(t, mErrOK)
 	})
 }
+
+func requireMultierrorNotContaining(t *testing.T, err error, unexpected ...string) {
+	t.Helper()
+	require.Error(t, err)
+	mErr, ok := err.(*multierror.Error)
+	require.True(t, ok)
+
+	var found []string
+	for _, unex := range unexpected {
+		if strings.Contains(mErr.Error(), unex) {
+			found = append(found, unex)
+		}
+	}
+
+	require.Empty(t, found, "found unexpected error(s): %v", found)
```

Comment:
I think this failure would output the found list twice.  I'd suggest cutting removing the second reference.

```suggestion
	require.Empty(t, found, "found unexpected error(s)")
```